libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -511,12 +537,18 @@ private Optional<JsonNode> generateAddressClaim(List<VcStoreItem> vcStoreItems)
         return Optional.of(addressNode);
     }
 
-    private Optional<JsonNode> generateNinoClaim(List<VcStoreItem> successfulVCStoreItems)
+    private Optional<JsonNode> generateNinoClaim(
+            List<VcStoreItem> successfulVCStoreItems, ProfileType profileType)
             throws HttpResponseExceptionWithErrorBody {
-        var ninoStoreItem = findStoreItem(NINO_CRI, successfulVCStoreItems);
+        String criToExtractFrom =
+                profileType.equals(ProfileType.GPG45) ? NINO_CRI : HMRC_MIGRATION_CRI;
+        var ninoStoreItem = findStoreItem(criToExtractFrom, successfulVCStoreItems);
 
         if (ninoStoreItem.isEmpty()) {
-            LOGGER.warn(LogHelper.buildLogMessage("Failed to find Nino CRI credential"));
+            LOGGER.warn(
+                    LogHelper.buildLogMessage(
+                            "Failed to find appropriate CRI credential to extract "
+                                    + NINO_PROPERTY_NAME));
```

Comment:
I'm not convinced these should be warnings tbh - it's perfectly valid to have a journey that doesn't collect a NINO (or passport or driving licence).

Not really part of your ticket so I'm also happy to leave as-is.

Reply:
We can change it to info....msg says that we didn't find appropriate (i.e nino / hmrcMigration) vc from the vc list to extract 'socialSecurityRecord'.